text/3502-cargo-script.md

```diff
@@ -0,0 +1,1206 @@
+- Feature Name: `cargo-script`
+- Start Date: 2023-03-31
+- Pre-RFC: [internals](https://internals.rust-lang.org/t/pre-rfc-cargo-script-for-everyone/18639)
+- eRFC PR: [rust-lang/rfcs#3424](https://github.com/rust-lang/rfcs/pull/3424)
+- RFC PR: [rust-lang/rfcs#3502](https://github.com/rust-lang/rfcs/pull/3502)
+- Rust Issue: [rust-lang/cargo#12207](https://github.com/rust-lang/cargo/issues/12207)
+
+# Summary
+[summary]: #summary
+
+This RFC adds support for single-file bin packages in cargo.
+Single-file bin packages are `.rs` files with an embedded manifest and a
+`main`.
+These will be accepted with just like `Cargo.toml` files with
```

Comment:
@ggggggggg 
> "These will be accepted with just like Cargo.toml files" doesn't read right.

I've moved this thread to the context to make it easier to follow.

Could you clarify why you feel this isn't right?

Reply:
"This RFC adds support for single-file bin packages in cargo. Single-file bin packages are .rs files with an embedded manifest and a main. cargo will be modified to accept cargo <file>.rs as a shortcut to cargo run --manifest-path <file>.rs; this allows placing cargo in a #! line for directly running these files."

Above I've deleted the 2nd sentence. The paragraph communicates clearly to me the goal. I have no idea what the 2nd sentence is trying to add, but specifically the phrase "with just like X" makes no sense.

Reply:
I've pushed a change that tries to clarify that sentence:

> These files will be accepted by cargo commands as `--manifest-path` just like `Cargo.toml` files.

Removing things isn't always the best solution when they aren't clear.  This isn't focused on some kind of cargo internals but core aspects of how you interact with it.  This is saying that these files are interchangeable with `Cargo.toml` and you can run commands like `cargo test --manifest-path foo.rs`.